libr/bin/format/elf/elf.c

```diff
@@ -348,6 +348,97 @@ static int init_dynamic_section (struct Elf_(r_bin_elf_obj_t) *bin) {
 	return true;
 }
 
+static RBinElfSection* get_section_by_name(struct Elf_(r_bin_elf_obj_t) *bin, const char *section_name) {
+	int i;
+	if (!g_sections) return NULL;
+	for (i = 0; !g_sections[i].last; i++) {
+		if (!strncmp (g_sections[i].name, section_name, ELF_STRING_LENGTH-1))
+			return &g_sections[i];
+	}
+	return NULL;
+}
+
+//static int offset_from_vaddr(struct Elf_(r_bin_elf_obj_t) *bin, int size) {
+//}
+
+static void store_versioninfo_gnu_versym(struct Elf_(r_bin_elf_obj_t) *bin, Elf_(Shdr) *shdr) {
+	int i;
+	const char *section_name = &bin->shstrtab[shdr->sh_name];
```

Comment:
check shdr->sh_name and other against the boundaries otherwise memory corruption might happen. However, i don't see the whole picture and maybe is checked before to come here.

Actually with a corrupted elf binary, `shdr` may not even be created as well as `shstrtab`. I suggest try r2r (https://github.com/radare/radare2-regressions) and use some elf binaries there (bins/elf/zzufed) or fuzz a little bit this feature to be sure and add some test would be welcomed.